onnxruntime/contrib_ops/cpu/bert/attention.cc

```diff
@@ -73,10 +74,13 @@ Status AttentionBase::CheckInputs(const TensorShape& input_shape,
   int batch_size = static_cast<int>(dims[0]);
   int sequence_length = static_cast<int>(dims[1]);
   int hidden_size = static_cast<int>(dims[2]);
-  if (hidden_size % num_heads_ != 0) {
-    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
-                           "Input 0 dimension 2 should be divisiable by value of the num_heads attribute.");
-  }
+  if (head_size_ < 0)
+    head_size_ = (int) hidden_size / num_heads_;
+  // This is not true for head-pruned transformers - e.g. FastFormers, 12 heads -> 7 heads
+  // if (hidden_size % num_heads_ != 0) {
+  //   return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+  //                          "Input 0 dimension 2 should be divisiable by value of the num_heads attribute.");
+  // }
```

Comment:
for the case head_size_ <0, we should still check this, right

Reply:
Yes, that's right. Will update it. Thanks!